### ISSUE TITLE
Blocking googlesyndication.com causes ##ins.adsbygoogle[data-ad-slot] not being applied

### ISSUE THREAD

**Issue by Yuki2718** [uBlockOrigin/uBlock-issues#3203]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

As the title says.

### A specific URL where the issue occurs.

```yaml
https://webtan.impress.co.jp/n/2017/05/12/25720
```

### Steps to Reproduce

1. Visit the site and see this leftover which should be partly hidden. Will be the same even if you add `##ins.adsbygoogle[data-ad-slot]` to My filters.

![leftover](https://github.com/uBlockOrigin/uBlock-issues/assets/58900598/41fc3b92-479e-401b-a111-65ce694618bb)

2. ~~Disable all the three lists~~ Unblock `googlesyndication.com` by dynamic rule and add `##ins.adsbygoogle[data-ad-slot]` if not yet done, refresh the page and see now the placeholder is partly hidden.

![hidden](https://github.com/uBlockOrigin/uBlock-issues/assets/58900598/27605f35-1dc1-4348-a1f5-3544ed4f50b5)

### Expected behavior

`##ins.adsbygoogle[data-ad-slot]` should be applied as no `$generichide` or whatsoever

### Actual behavior

Described

### uBO version

1.57.0

### Browser name and version

Chrome 123.0.6312.106 and Firefox 124.0.2

### Operating System and version

Windows 10

**Comment by gwarser** [comment]
Requires `:watch-attr()`.

**Comment by gwarser** [comment]
Also does not work. No idea.

When element is visible, so it's not `_ph`, after reload the filter from EasyList is applied.

**Comment by Yuki2718** [comment]
Yes, works if specific. So maybe the problem is in DOM surveyor?

**Comment by gorhill** [comment]
The surveyor does not trigger on attribute changes, only on layout changes. View-source shows that the class name is `adsbygoogle_ph`:

```html
<div id="bancho_adsense_nodefooter" style="margin:0; display:flex; flex-wrap:wrap; justify-content:space-around; gap:1em 0; padding:.5em 0; overflow:hidden">
    <div style="display:inline-block; width:300px; max-width:300px; margin:0; overflow:hidden">
        <ins class="adsbygoogle_ph" style="display:inline-block;width:300px;height:250px" data-ad-client="ca-pub-4894439427301282" data-ad-slot="1000693078" data-ad-format="auto"></ins>
    </div>
    <div style="display:inline-block; width:300px; max-width:300px; margin:0; overflow:hidden">
        <ins class="adsbygoogle_ph" style="display:inline-block;width:300px;height:250px" data-ad-client="ca-pub-4894439427301282" data-ad-slot="2477426276" data-ad-format="auto"></ins>
    </div>
</div>
```

It works if relying on the attribute which does not change:

    ##ins[data-ad-slot]

Or when using specific filter:

    impress.co.jp##ins.adsbygoogle[data-ad-slot]

I consider this is a duplicate of <https://github.com/gorhill/uBlock/issues/1885>, see [my comment](https://github.com/gorhill/uBlock/issues/1885#issuecomment-877655557) closing the issue.

**Comment by Yuki2718** [comment]
All right, I'll ask filter authors if we can change the rule to `##ins[data-ad-slot]`.

**Comment by gorhill** [comment]
> All right, I'll ask filter authors if we can change the rule to `##ins[data-ad-slot]`

Not worth risking breaking what works well now for the sake of one case found which requires an easy workaround.
